src/Passes/Source/AllocationManager/AllocationManager.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "AllocationManager/AllocationManager.hpp"
+
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    BasicAllocationManager::BasicAllocationManagerPtr BasicAllocationManager::createNew()
+    {
+        BasicAllocationManagerPtr ret;
+        ret.reset(new BasicAllocationManager());
```

Comment:
FYI: `std::make_shared` can be more effective. See [Item 21](https://moodle.ufsc.br/pluginfile.php/2377667/mod_resource/content/0/Effective_Modern_C__.pdf).

Reply:
`std::make_shared` does not work here because the constructor is private. This is deliberate by design as it ensures that AllocationManager can only be allocated as a shared pointer.

Reply:
Thanks for explanation, makes total sense.
(In such cases, for my own code I typically place a comment (e.g. "`std::make_shared` does not work here because the constructor is private") briefly explaining the decision, to immediately resolve any questions or comments that the readers can have, like the comment I started this discussion with)
Thank you. No need to change the code.